### ISSUE TITLE
Low FPS on frames from buffers, when try to move

### ISSUE THREAD

**Issue by basil-ionin** [mwydmuch/ViZDoom#252]
Hello!

I have a problem: when I don't push any buttons and stay in one place, I got very nice FPS in frames generated via screenBuffer/depthBuffer, but when I'm moving in some way, FPS of generated frames is very bad. I have two separate threads: in first I generate frames, in second I'm waiting for keys and display frames in window. You can look at my code here: https://gist.github.com/basil-ionin/d4a913843dd1a38d85f1103d429d4f85.

Also, I recorded a small video fragment, when you can see my problem: https://youtu.be/Vb1YaKIAQds.

For displaying images I use OpenCV. I have tried to execute my program in VisualStrudio Debug mode and Release (both in Multithreading Runtime), but I got the same result.

P. S. Sorry for my English

**Comment by basil-ionin** [comment]
Sorry, it was thread trouble :(

**Comment by mwydmuch** [comment]
Hi @basil-ionin, no at all.
Why did you close the issue? Have you found the reason for this behavior?

**Comment by basil-ionin** [comment]
No, I started to display images in the moment, when I generate it, the FPS has increased (but this is a bad solution, because, I can't segregate duties to nodes, and this is very strange), so I can handle it now. But it still lag a little bit and I just recognise, that I can't handle real fightings, because of the velocity of enemies :(
